Git-commit: 87424dc73510bebbcddaa80f6e9c864394865f1f
From: Hristo Venev <hristo@venev.name>
Date: Wed, 19 Aug 2020 17:05:53 +0300
Subject: [PATCH] staging: media: rpivid: Select MEDIA_CONTROLLER and
 MEDIA_CONTROLLER_REQUEST_API
Patch-mainline: Queued in openEuler repo, version 5.10.0-4.5.0
References: openEuler-21.03
Git-repo: https://gitee.com/openeuler/kernel.git

raspberrypi inclusion
category: feature
bugzilla: 50432

--------------------------------

MEDIA_CONTROLLER_REQUEST_API is a hidden option. If rpivid depends on it,
the user would need to first enable another driver that selects
MEDIA_CONTROLLER_REQUEST_API, and only then rpivid would become available.

By selecting it instead of depending on it, it becomes possible to enable
rpivid without having to enable other potentially unnecessary drivers.

Signed-off-by: Hristo Venev <hristo@venev.name>
Signed-off-by: Fang Yafen <yafen@iscas.ac.cn>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Kai Liu <kai.liu@suse.com>
---
 drivers/staging/media/rpivid/Kconfig | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/staging/media/rpivid/Kconfig b/drivers/staging/media/rpivid/Kconfig
index 58b470c80db1..304c3edf0e71 100644
--- a/drivers/staging/media/rpivid/Kconfig
+++ b/drivers/staging/media/rpivid/Kconfig
@@ -3,9 +3,9 @@
 config VIDEO_RPIVID
 	tristate "Rpi H265 driver"
 	depends on VIDEO_DEV && VIDEO_V4L2
-	depends on MEDIA_CONTROLLER
 	depends on OF
-	depends on MEDIA_CONTROLLER_REQUEST_API
+	select MEDIA_CONTROLLER
+	select MEDIA_CONTROLLER_REQUEST_API
 	select VIDEOBUF2_DMA_CONTIG
 	select V4L2_MEM2MEM_DEV
 	help
-- 
2.31.1

